ilastik/experimental/parser/types.py

```diff
@@ -23,30 +34,87 @@ class FeatureMatrix:
+
+        feature_names = [name.decode("ascii") for name in features_group[Keys.FEATURES_IDS][()]]
+        scales = features_group[Keys.FEATURES_SCALES][()]
+        sel_matrix = features_group[Keys.FEATURES_SELECTION_MATRIX][()]
+        compute_in_2d = features_group[Keys.FEATURES_COMPUTE_IN_2D][()]
+
+        return FeatureMatrix(feature_names, scales, sel_matrix, compute_in_2d)
+
 
 @dataclass
 class Classifier:
     instance: Any
     factory: Any
     label_count: int
 
+    @staticmethod
+    def from_ilp_group(pixel_class_group: h5py.Group) -> "Classifier":
+        class Keys(StrEnum):
+            PIXEL_CLASSIFICATION_FORESTS = "ClassifierForests"
+            PIXEL_CLASSIFICATION_FACTORY = "ClassifierFactory"
+            PIXEL_CLASSIFICATION_TYPE = "ClassifierForests/pickled_type"
+            LABEL_NAMES = "LabelNames"
+
+        if missingkeys := [k for k in Keys if k not in pixel_class_group]:
+            raise IlastikAPIError(f"Missing keys in project file: {missingkeys}")
+
+        classifier_type = pickle.loads(pixel_class_group[Keys.PIXEL_CLASSIFICATION_TYPE][()])
+        classifier = classifier_type.deserialize_hdf5(pixel_class_group[Keys.PIXEL_CLASSIFICATION_FORESTS])
+
+        classifier_factory = pickle.loads(pixel_class_group[Keys.PIXEL_CLASSIFICATION_FACTORY][()])
+        label_count = len(pixel_class_group[Keys.LABEL_NAMES])
+
+        return Classifier(classifier, classifier_factory, label_count)
+
 
 @dataclass
 class ProjectDataInfo:
     spatial_axes: str
     num_channels: int
     axis_order: str
 
+    @staticmethod
+    def from_ilp_group(infos_group: h5py.Group) -> "ProjectDataInfo":
+        class Keys(StrEnum):
+            # input data keys in project file
+            RAW_DATA = "Raw Data"
+            RAW_DATA_AXIS_TAGS = "Raw Data/axistags"
+            RAW_DATA_SHAPE = "Raw Data/shape"
+
+        info = next(iter(infos_group.values()))
+
+        if missingkeys := [k for k in Keys if k not in info]:
+            raise IlastikAPIError(f"Missing data info keys in project file: {missingkeys}.")
+
+        json_bytes = info[Keys.RAW_DATA_AXIS_TAGS][()]
+        tags_dict = json.loads(json_bytes.decode("ascii"))
+        shape = info[Keys.RAW_DATA_SHAPE][()]
+
+        if len(shape) != len(tags_dict["axes"]):
+            raise IlastikAPIError(f"Shape {shape} and axistags {tags_dict} mismatch")
 
-class PixelClassificationProject(abc.ABC):
-    @abc.abstractproperty
-    def feature_matrix(self) -> Optional[FeatureMatrix]:
-        ...
+        spatial_axes = ""
+        axis_order = ""
+        num_channels = 1
+        for size, dim in zip(shape, tags_dict["axes"]):
+            if dim["key"] in "xyz":
+                spatial_axes += dim["key"]
+            elif dim["key"] == "c":
+                num_channels = size
 
-    @abc.abstractproperty
-    def classifier(self) -> Optional[Classifier]:
-        pass
+            axis_order += dim["key"]
```

Comment:
```python
axis_order = "".join(dim["key"] for dim in tags_dict["axes"])
spatial_axes = "".join(axis for axis in axis_order if axis in "xyz")
try:
    num_channels = shape[axis_order.index("c")]
except ValueError:
    num_channels = 1
```

Reply:
looks good, for spacial axes i'll opt for

```Python
spatial_axes = "".join(ax for ax in axis_order if ax in "xyz")
```
for subjectively better readability.